### ISSUE TITLE
astype('str') behavior is different in 1.6.0 vs 1.5.1 (migrated from Trac #1748)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3298]
Original ticket http://projects.scipy.org/numpy/ticket/1748
Reported 2011-02-21 by atmention:jseabold, assigned to unknown.

```
>>> import numpy as np
>>> np.__version__
'1.6.0.dev-c50af53'
>>> tmp_arr = np.array([['black'],['white'],['other']])
>>> tmp_arr.astype('str')
array([['b'],
      ['w'],
      ['o']],
     dtype='|S1')
```

The old behavior

```
>>> import numpy as np
>>> np.__version__
'1.5.1'
>>> tmp_arr = np.array([['black'],['white'],['other']])
>>> tmp_arr.astype('str')
array([['black'],
      ['white'],
      ['other']],
     dtype='|S5')
```

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-03-03

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-03-07

Fixed in cfff7508bc29e6bc0c44.
